src/v/pandaproxy/schema_registry/seq_writer.h

```diff
@@ -0,0 +1,115 @@
+//// Copyright 2021 Vectorized, Inc.
+////
+//// Use of this software is governed by the Business Source License
+//// included in the file licenses/BSL.md
+////
+//// As of the Change Date specified in that file, in accordance with
+//// the Business Source License, use of this software will be governed
+//// by the Apache License, Version 2.
+
+#pragma once
+
+#include "kafka/client/client.h"
+#include "pandaproxy/schema_registry/sharded_store.h"
+#include "pandaproxy/schema_registry/types.h"
+#include "random/simple_time_jitter.h"
+#include "utils/retry.h"
+
+namespace pandaproxy::schema_registry {
+
+using namespace std::chrono_literals;
+
+static const int max_retries = 32;
+
+class seq_writer final : public ss::peering_sharded_service<seq_writer> {
```

Comment:
I don't understand why this class is a `peering_sharded_service`, only the state on core0 is used.

Reply:
> the peering sharded service allowed the removal of prolific smp::submit_to calls.

4 calls to `submit_to` have become 4 calls to `invoke_on`, which is basically the same thing.
```cpp
invoke_on(unsigned id, smp_submit_to_options options, Func&& func, Args&&... args) {
    return smp::submit_to(id, options, [this, func = std::forward<Func>(func), args = std::tuple(std::move(args)...)] () mutable {
        auto inst = get_local_service();
        return std::apply(std::forward<Func>(func), std::tuple_cat(std::forward_as_tuple(*inst), std::move(args)));
    });
}
```
But now instead of having the state once, in the startup thread, now it's on each core, but only used on core 0.

> but I agree @BenPope I think this could be a simple object stored in a unique_ptr inside some parent service on core 0 with some some well defined central dispatching to core 0 (what we do right now for the controller service--perhaps still using the low-level submit_to).

I'm not sure what's not well defined about `submit_to`.

> perhaps we can do that as a follow-up refactor?

Yeah, I won't block this.